packages/rest/src/lib/CDN.ts

```diff
@@ -11,39 +11,70 @@ import {
 } from './utils/constants.js';
 
 /**
- * The options used for image URLs
+ * The options used for image URLs.
  */
 export interface BaseImageURLOptions {
 	/**
-	 * The extension to use for the image URL
+	 * The extension to use for the image URL.
 	 *
 	 * @defaultValue `'webp'`
 	 */
 	extension?: ImageExtension;
 	/**
-	 * The size specified in the image URL
+	 * The size specified in the image URL.
 	 */
 	size?: ImageSize;
 }
 
+export interface EmojiURLOptionsWebp extends BaseImageURLOptions {
+	/**
+	 * Whether to use the `animated` query parameter.
+	 */
+	animated?: boolean;
+	extension?: 'webp';
+}
+
+export interface EmojiURLOptionsNotWebp extends BaseImageURLOptions {
+	extension: Exclude<ImageExtension, 'webp'>;
+}
+
 /**
- * The options used for image URLs with animated content
+ * The options used for emoji URLs.
  */
-export interface ImageURLOptions extends BaseImageURLOptions {
+export type EmojiURLOptions = EmojiURLOptionsNotWebp | EmojiURLOptionsWebp;
+
+export interface BaseAnimatedImageURLOptions extends BaseImageURLOptions {
 	/**
-	 * Whether or not to prefer the static version of an image asset.
+	 * Whether to prefer the static asset.
 	 */
 	forceStatic?: boolean;
 }
 
+export interface ImageURLOptionsWebp extends BaseAnimatedImageURLOptions {
+	extension?: 'webp';
+}
+
+export interface ImageURLOptionsNotWebp extends BaseAnimatedImageURLOptions {
+	extension: Exclude<ImageExtension, 'webp'>;
+}
+
+/**
+ * The options used for image URLs that may be animated.
+ */
+export type ImageURLOptions = ImageURLOptionsNotWebp | ImageURLOptionsWebp;
+
```

Comment:
What's the purpose of this split? It doesn't seem like either of the union members gets used separately anywhere.

Reply:
Redundant code from earlier commits. Resolved with 2b55f82501f3bf48baa92e48fd47090586d22e92.